### PR TITLE
WT-6846 Initial test program for the new cpp test framework.

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

File: test/cppsuite/tests/poc.cxx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#include <iostream>
+extern "C" {
+    #include "wiredtiger.h"
+}
+
+int main(int argc, char *argv[]) {
+    WT_CONNECTION *conn;
+    int ret = 0;
+    // Setup basic test directory.
+    std::string default_dir = "WT_TEST";
```
Comment (tetsuo-cpp) [inline test/cppsuite/tests/poc.cxx:10]: The `default_dir` and `mkdir_cmd` strings can be marked as const.

**Review by tetsuo-cpp — APPROVED** [review]
LGTM.

File: test/cppsuite/tests/poc.cxx (hunk 1 above)
Comment (luke-pearson) [reply test/cppsuite/tests/poc.cxx]: 4a87eda

File: test/cppsuite/tests/poc.cxx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    #include "wiredtiger.h"
+}
+
+int main(int argc, char *argv[]) {
+    WT_CONNECTION *conn;
+    int ret = 0;
+    // Setup basic test directory.
+    const std::string default_dir = "WT_TEST";
+
+    // Csuite tests utilise a test_util.h command to make their directory, currently that doesn't
```
Comment (sulabhM) [inline test/cppsuite/tests/poc.cxx:12]: Is the intention here to use the _C++ style_ comments. I would suggest we continue using the coding conventions we have for the rest of the wired tiger code.

**Review by sulabhM — APPROVED** [review]
Comment on comment style, otherwise LGTM

File: test/cppsuite/tests/poc.cxx (hunk 2 above)
Comment (luke-pearson) [reply test/cppsuite/tests/poc.cxx:12]: Oh duh, yeah I'll fix that up.

**Pull request merged**